conanfile.py

```diff
@@ -1,54 +1,51 @@
-from conans import ConanFile, CMake, tools
+import os, pathlib
+
+from conan import ConanFile
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools.files import patch, copy
+from conan.tools.scm import Git
 
 
 class FMILibraryConan(ConanFile):
     name = "fmilibrary"
     version = "2.3"
     license = "https://github.com/modelon-community/fmi-library/blob/master/LICENSE.md"
-    url = "https://github.com/kyllingstad/conan-FMILibrary"
+    url = "https://github.com/open-simulation-platform/conan-fmilibrary"
     description = "An implementation of the FMI standard which enables FMU import in applications"
-    scm = {
-        "type": "git",
-        "url": "https://github.com/modelon-community/fmi-library.git",
-        "revision": version,
-        "subfolder": "src"
-    }
+    
     settings = "os", "compiler", "build_type", "arch"
     options = {"shared": [True, False]}
-    default_options = "shared=False"
-    generators = "cmake"
-    exports_sources = [ "build-static-c99snprintf.patch" ]
+    default_options = {"shared": False}
+
+    generators = "CMakeDeps"
+
+    exports_sources = "build-static-c99snprintf.patch"
 
     def source(self):
-        tools.patch(base_path="src", patch_file="build-static-c99snprintf.patch")
-        tools.replace_in_file("src/CMakeLists.txt", "project (FMILibrary)",
-                              '''project (FMILibrary)
-include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-conan_basic_setup()''')
+        git = Git(self)
+        git.clone(url="https://github.com/modelon-community/fmi-library.git", target="src", args=["--branch=2.3"])
+        patch(self, base_path="src", patch_file="build-static-c99snprintf.patch")
+
+    def layout(self):
+        cmake_layout(self)
 
+    def generate(self):       
+        tc = CMakeToolchain(self)
+        tc.cache_variables["FMILIB_BUILD_STATIC_LIB"] = False if self.options.shared else True
```

Comment:
I believe these should be `variables` instead of `cache_variables`

Reply:
Are you sure? According to [the `variables` documentation](https://docs.conan.io/2/reference/tools/cmake/cmaketoolchain.html#variables),

> This attribute allows defining CMake variables, for multiple configurations (Debug, Release, etc). These variables should be used to define things related to the toolchain and for the majority of cases cache_variables is what you probably want to use.